Components and Props translation (#9)
reactjs/hu.reactjs.org: content/docs/components-and-props.md

next: state-and-lifecycle.html
---

> A komponensek lehetővé teszik számodra a felhasználói felület független, újrafelhasználható darabokra való felosztását, és segítnek hogy minden darabról a többitől elzártan tudj gondolkodni. Ez az oldal a komponensek lényegét mutatja be. A [részletes komponens API referenciát itt](/docs/react-component.html) találod.

Reviewer: ```suggestion
A komponensek lehetővé teszik számodra a felhasználói felület független, újrafelhasználható darabokra való felosztását, és segítenek hogy minden darabról a többitől elzártan tudj gondolkodni. Ez az oldal a komponensek lényegét mutatja be. A [részletes komponens API referenciát itt](/docs/react-component.html) találod.
```

Elviekben a komponensek olyanok mint a JavaScript függvények. Egy tetszőleges számú inputot fogadnak (amiket "prop"-oknak hívunk) és egy React elemet adnak vissza ami leírja mi jelenjen meg a képernyőn.

> ## Függvény és Osztály komopnensek {#function-and-class-components}

Reviewer: ```suggestion
## Függvény és Osztály komponensek {#function-and-class-components}
```

[](codepen://components-and-props/rendering-a-component)

Foglaljuk össze mi történik ebben a példában:

1. Meghívjuk a `ReactDOM.render()` metódust a `<Welcome name="Sára" />` elemmel.
2. A React meghívja a `Welcome` komponenst a `{name: 'Sára'}` props objektummal.

> 3. A `Welcome` komponensünk visszatéríti a `<h1>Helló, Sára</h1>` elemet eredményként.

Reviewer: ```suggestion
3. A `Welcome` komponensünk visszaadja a `<h1>Helló, Sára</h1>` elemet eredményként.
```

[](codepen://components-and-props/extracting-components)

Ez fogad egy `author` (objektumot), `text` (karakterláncot), és `date` (dátumot) props-ként, és egy kommentet ír le egy közösségi média weblapon.

> Ezt a komponenst furfangos lehet megváltoztatni a sok egymásba ágyazás miatt, és nehéz is újrafelhasználni az egyedülálló részeit. Vonjunk ki egy pár komponenst belőle.

Reviewer: ```suggestion
Ezt a komponenst furfangos lehet megváltoztatni a sok egymásba ágyazás miatt, és nehéz is újra felhasználni az egyedülálló részeit. Vonjunk ki egy pár komponenst belőle.
```

[](codepen://components-and-props/extracting-components-continued)

A komponensek kivonása elsőre morgós munkának tűnhet, de nagyobb alkalmazások esetén gyorsan megtérül ha egy újrafelhasználható komponens palettával rendelkezünk. Egy jó ökölszabály ha a felhasználói kezelőfelületed valamelyik része többször fel van használva (`Button`, `Panel`, `Avatar`), vagy elég bonyolult saját magában is (`App`, `FeedStory`, `Comment`) akkor jó jelölt lehet egy újrafelhasználható komponensnek.

## A prop-ok csak olvashatók {#props-are-read-only}

Függetlenül hogy egy komponenst [függvényként vagy osztályként](#function-and-class-components) deklarálsz, az soha nem módosíthatja annak saját prop-jait. Vedd ezt a `sum` függvényt:

```js
function sum(a, b) {
  return a + b;
}
```

Egy ilyen függvényt ["tiszta"](https://hu.wikipedia.org/wiki/Tiszta_függvény) függvénynek nevezünk, mert nem kísérli meg megváltoztatni a bemenetét, és mindig ugyanazt az eredményt adja ugyanazon bemenet esetében.

> Összehasonlítás képpen ez a függvény nem tiszta, mert megváltoztatja a saját bemenetét:

Reviewer: ```suggestion
Összehasonlításképpen ez a függvény nem tiszta, mert megváltoztatja a saját bemenetét:
```
